### ISSUE TITLE
Rsync to goofys mounted S3 Bucket creates locks & hangs

### ISSUE THREAD

**Issue by painslie** [minio/minio#5143]
I mounted a Minio bucket via goofy and proceeded to rsync a folder of 45,000 files (1.5 TB).  It got as far as copying 11,000 files and then hung.

The Minio cluster
 - is in Germany and the goofys mount is in London, hence a 16 ms latency
 - is distributed and consists of 4 servers, 8 hard drives.
 - the hard drives are mechanical  ;(

`$ mc admin lock list --duration 1m minio-001/71026777`

```71026777/vod/YYstV-0CpNf.mp4 (LocksOnObject: 0, locksAcquiredOnObject: 0, totalBlockLocks:0):   {OperationID:LYD6861UZ948D64N LockSource:[object-handlers.go:354:objectAPIHandlers.CopyObjectHandler()] LockType:WLock Status:Running Since:2017-10-26 11:59:22.880470396 +0000 UTC}

71026777/YYstV-0CpNf.mp4 (LocksOnObject: 0, locksAcquiredOnObject: 0, totalBlockLocks:0):   {OperationID:AHAJM84NKF1029YK LockSource:[object-handlers.go:367:objectAPIHandlers.CopyObjectHandler()] LockType:RLock Status:Running Since:2017-10-26 11:59:22.881767679 +0000 UTC}

71026777/YYstV-0CpNf.mp4 (LocksOnObject: 0, locksAcquiredOnObject: 0, totalBlockLocks:0):   {OperationID:AHAJM84NKF1029YK LockSource:[object-handlers.go:367:objectAPIHandlers.CopyObjectHandler()] LockType:RLock Status:Running Since:2017-10-26 11:59:22.881767679 +0000 UTC}  {OperationID:0GR8X7CC4YS9A154 LockSource:[object-handlers.go:367:objectAPIHandlers.CopyObjectHandler()] LockType:RLock Status:Running Since:2017-10-26 14:04:01.553397179 +0000 UTC}

71026777/YYstV-0CpNf-1496029126.mp4 (LocksOnObject: 0, locksAcquiredOnObject: 0, totalBlockLocks:0):   {OperationID:1RL1HBQGK3O3N2D7 LockSource:[object-handlers-common.go:235:deleteObject()] LockType:WLock Status:Running Since:2017-11-02 13:17:55.416009735 +0000 UTC}
```

## Expected Behavior
This should have successfully synced all files after about 3 days.

## Current Behaviour
After about a day or so the rsync hung and I had to kill goofys 
## Possible Solution
Goofys 0.0.18 "Fixed a deadlock that can happen under memory pressure".  I'll try this again with current version.

## Steps to Reproduce (for bugs)
1. Run goofys 0.0.17
2. mount Minio bucket (running Minio )
3. rsync a folder with, like, 45,000 files (1.5 TB)
4. go to bed for a

## Context
I'm trying to sync up a large amount of video files to an S3 bucket.  I prefer goofys because it's the fastest, especially in terms of `ls` type commands.

## Your Environment
* Version used:
```Version: 2017-10-19T09:45:15Z
Release-Tag: DEVELOPMENT.2017-10-19T09-45-15Z
Commit-ID: e65d8fdc7d3cf5492a99313532b512c71eac5289
```
* Environment name and version:
goofys 0.0.17
* Server type and version: 
Intel Core i7-2600 3.80 GHz
16 GB RAM 
2x 3 TB GB HD's
* Operating System and version (`uname -a`):
Debian 8
4.10.0-35-generic #39~16.04.1-Ubuntu SMP Wed Sep 13 09:02:42 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**Comment by harshavardhana** [comment]
> Release-Tag: DEVELOPMENT.2017-10-19T09-45-15Z
> Commit-ID: e65d8fdc7d3cf5492a99313532b512c71eac5289

Since this looks like a development build would it possible for you to test with latest release? `minio.RELEASE.2017-10-27T18-59-02Z`  at https://dl.minio.io/server/minio/release/linux-amd64/

**Comment by kahing** [comment]
it does sound like this could be a goofys bug fixed in 0.0.18

**Comment by harshavardhana** [comment]
> it does sound like this could be a goofys bug fixed in 0.0.18

Thanks @kahing @painslie has not confirmed yet if migrating to 0.0.18 has fixed the bug.

**Comment by vadmeste** [comment]
Closing this due to inactivity @painslie, please open again if you see this problem again.
